### PR TITLE
[IMP] account_chart_update: update taxes when children are differents

### PR DESCRIPTION
**Pull request by cubells**
With these changes, account_chart_update will update taxes if their children are not the same than template ones.

updated spanish translation

cc @Tecnativa

### CODE REVIEW

File: account_chart_update/i18n/es.po [hunk 1]
```diff
@@ -1,73 +1,79 @@
...
 #: model:ir.ui.view,arch_db:account_chart_update.view_update_multi_chart
 msgid "Chart of Accounts"
-msgstr ""
+msgstr "Plan contable"
+
+#. module: account_chart_update
+#: code:addons/account_chart_update/wizard/wizard_chart_update.py:460
+#, python-format
+msgid "Children taxes are not the same."
+msgstr "los impuestos hijos no son los mismos."
```
Comment (pedrobaeza) [inline account_chart_update/i18n/es.po:71]: Upper

File: account_chart_update/i18n/es.po [hunk 2]
```diff
@@ -112,40 +118,42 @@ msgid "Created on"
...
-msgstr ""
+msgstr "Impuestos desactivados"
 
 #. module: account_chart_update
 #: model:ir.model.fields,help:account_chart_update.field_wizard_update_charts_accounts_tax_type_tax_use
 msgid ""
 "Determines where the tax is selectable. Note : 'None' means a tax can't be "
 "used by itself, however it can still be used in a group."
 msgstr ""
+"Determina si el impuesto es seleccionable. Note: 'Ninguno' significa que un "
```
Comment (pedrobaeza) [inline account_chart_update/i18n/es.po:149]: s/Note/Nota

File: account_chart_update/i18n/es.po [hunk 3]
```diff
@@ -156,62 +164,64 @@ msgid "Display Name"
...
 #, python-format
 msgid "Exception writing account %s."
-msgstr ""
+msgstr "Excepción al escribir la cuenta %s."
 
 #. module: account_chart_update
 #: model:ir.model.fields,help:account_chart_update.field_wizard_update_charts_accounts_update_account
 msgid "Existing accounts are updated. Accounts are searched by code."
-msgstr ""
+msgstr "Las cuentas exitentes están actualizadas. Las cuentas se han buscado por código."
```
Comment (pedrobaeza) [inline account_chart_update/i18n/es.po:181]: s/exitentes/existentes

File: account_chart_update/i18n/es.po [hunk 4]
```diff
@@ -227,23 +237,28 @@ msgid ""
 "If set, the wizard will continue to the next step even if there are minor "
 "errors."
 msgstr ""
+"Si está marcado, el asistente continuará al siguiente paso incluos si hay "
```
Comment (pedrobaeza) [inline account_chart_update/i18n/es.po:240]: s/incluos/incluso

File: account_chart_update/i18n/es.po [hunk 5]
```diff
@@ -227,23 +237,28 @@ msgid ""
...
 
 #. module: account_chart_update
 #: model:ir.ui.view,arch_db:account_chart_update.view_update_multi_chart
 msgid ""
-"If you leave these options set, the wizard will not just create new records,"
-" but also update records with changes (i.e. different tax amount)"
+"If you leave these options set, the wizard will not just create new records, "
+"but also update records with changes (i.e. different tax amount)"
 msgstr ""
+"Si deja estas opciones marcadas, el asistente no solamente crearé los nuevos "
```
Comment (pedrobaeza) [inline account_chart_update/i18n/es.po:249]: s/crearé/creará

File: account_chart_update/i18n/es.po [hunk 6]
```diff
@@ -380,147 +400,150 @@ msgstr "Estado"
...
+msgstr "Impuestos"
 
 #. module: account_chart_update
 #: model:ir.ui.view,arch_db:account_chart_update.view_update_multi_chart
 msgid ""
 "This wizard will update your accounts, taxes and fiscal positions according "
 "to the selected chart template"
 msgstr ""
+"Este asistente actualizará sus cuentas, impuestos y posiciones fiscales de "
+"acuerdo a la plan de cuentas seleccionado."
```
Comment (pedrobaeza) [inline account_chart_update/i18n/es.po:454]: s/a la/al

File: account_chart_update/i18n/es.po [hunk 7]
```diff
@@ -380,147 +400,150 @@ msgstr "Estado"
...
 #: model:ir.model.fields,field_description:account_chart_update.field_wizard_update_charts_accounts_tax_update_chart_wizard_id
 msgid "Update chart wizard"
-msgstr ""
+msgstr "Asistente de actualización de cuentas"
 
 #. module: account_chart_update
 #: model:ir.model.fields,field_description:account_chart_update.field_wizard_update_charts_accounts_update_fiscal_position
 msgid "Update fiscal positions"
-msgstr ""
+msgstr "Actualitzar posiciones fiscales"
```
Comment (pedrobaeza) [inline account_chart_update/i18n/es.po:495]: s/Actualitzar/Actualizar

File: account_chart_update/i18n/es.po [hunk 8]
```diff
@@ -380,147 +400,150 @@ msgstr "Estado"
...
 
 #. module: account_chart_update
 #: model:ir.ui.view,arch_db:account_chart_update.view_update_multi_chart
 msgid "Update records?"
-msgstr ""
+msgstr "Actualizar registros?"
```
Comment (pedrobaeza) [inline account_chart_update/i18n/es.po:500]: Opening question symbol

File: account_chart_update/wizard/wizard_chart_update.py [hunk 9]
```diff
@@ -456,6 +456,8 @@ def diff_notes(self, template, real):
         if template._name == "account.tax.template":
             if not real.active:
                 result.append(_("Tax is disabled."))
+            if real.children_tax_ids.ids != template.children_tax_ids.ids:
```
Comment (pedrobaeza) [inline account_chart_update/wizard/wizard_chart_update.py:459]: These IDs of course are not going to be the same, as one are taxes and the other are taxes templates. You have to use the same comparison as for parent taxes.

**Review by pedrobaeza — CHANGES_REQUESTED** [review]
Apart from the rest of the comments, change module version and include tests.

**Comment by pedrobaeza** [comment]
@cubells please rebase, resolve conflicts and attend commits.

**Comment by sbidoul** [comment]
At line 375, account_chart_update ignore child taxes. Do you know why it is so? @pedrobaeza

**Comment by pedrobaeza** [comment]
I think it's because each tax is evaluated independently, counting also their parent field, so children tax will be compared and if different parent, then present it to update. No need to check reverse one2many field.

**Comment by pedrobaeza** [comment]
Superseeded by #693

**Pull request closed without merging**